ci/bin/format.dart

```diff
@@ -786,6 +798,133 @@ class GnFormatChecker extends FormatChecker {
   }
 }
 
+/// Checks the format of any .dart files using the "dart format" command.
+class DartFormatChecker extends FormatChecker {
+  DartFormatChecker({
+    super.processManager,
+    required super.baseGitRef,
+    required Directory repoDir,
+    super.allFiles,
+    super.messageCallback,
+  }) : super(
+    repoDir: repoDir,
+  ) {
+    // $ENGINE/flutter/third_party/dart/tools/sdks/dart-sdk/bin/dart
```

Comment:
I think this is correct, but FYI this is the prebuilt dev channel Dart SDK, and *not* the prebuilt Dart SDK whose version matches the hash in the DEPS file.

Reply:
@zanderso For my own understanding: What's the difference between the two and where is the version of either of these specified?

Reply:
The versions are specified in the DEPS file:

The prebuilt dev channel SDK is here: https://github.com/flutter/engine/blob/4d001245329bc0f5db5f488b685e09dd03c5d76a/DEPS#L431

vs.

here we download a prebuilt Dart SDK that matches the version of the Dart SDK source that is pulled into the engine: https://github.com/flutter/engine/blob/4d001245329bc0f5db5f488b685e09dd03c5d76a/DEPS#L435

The version of the Dart SDK that the framework will be using is the second one.